src/lidar_subscriber.cpp

```diff
@@ -0,0 +1,156 @@
+
+#include <memory>
+#include <cmath>
+#include "rclcpp/rclcpp.hpp"
+
+#include "core-structs.h"
+
+
+#include "sensor_msgs/msg/laser_scan.hpp"
+#include <queue>
+#include <limits>
+#include "visualization_msgs/msg/marker.hpp"
+
+constexpr float LANDMARK_THRESHOLD_m = 0.02;
+constexpr bool testing = true;
+
+bool same_landmark(const Observation2D &first, const Observation2D &second){
+  float r1 = first.range_m;
+  float r2 = second.range_m;
+  float theta1 = first.bearing_rad;
+  float theta2 = second.bearing_rad;
+  float distance = sqrt(pow(r1,2.0)+pow(r2, 2.0)-2*r1*r2*cos(theta1-theta2));
+  if (distance >= LANDMARK_THRESHOLD_m){
+	//std::cout<<distance<<" ";		 
+   }
+  return  distance <= LANDMARK_THRESHOLD_m;
+}
+
+Observation2D merge(Observation2D &existing_landmark, Observation2D &new_measurement, int &count){
+  struct Observation2D average;
+  average.range_m = ((count-1)*existing_landmark.range_m+new_measurement.range_m)/count;
+  if(existing_landmark.bearing_rad>M_PI){
+	  existing_landmark.bearing_rad -=2*M_PI;
+  }
+  if(new_measurement.bearing_rad>M_PI){
+	  new_measurement.bearing_rad-=2*M_PI;
+  }
+  average.bearing_rad = ((count-1)*existing_landmark.bearing_rad+new_measurement.bearing_rad)/count;
+  return average;
+}
+
+Observation2D merge(Observation2D &first, Observation2D &second, const int &count1, const int &count2){
```

Comment:
`first` and `second` here are read-only, so make them `const` to protect against unintended changes.

Reply:
I actually do modify the bearings (in order to wrap the angle so it is in between pi and -pi), so it doesn't make sense to make them ```const```